### PR TITLE
BW-404: Significant reduction in cost to return workflow costs

### PR DESCRIPTION
**Pull request by kcibul**
Changes for [BW-404](https://broadworkbench.atlassian.net/browse/BW-404)

See corresponding change in [firecloud-devel](https://github.com/broadinstitute/firecloud-develop/pull/2325)

Technical details behind the optimization and benchmarks [here](https://docs.google.com/document/d/1190OjnqPuEWvkfBhowvdBqRsEaaW-JdJhVaE4aEeZS8/edit#)

As a test I also launched a FIAB, launched a new workflow, waited for the billing information to appear and confirmed it showed up correctly in the FIAB Terra UI

Open Questions
1. Use of bind parameters (?) in the queries was inconsistent.  Used for projectId, but not for submissionId.  I tried to follow what was there, but it would help to know the reason for not using bind parameters everywhere (sql injection) or nowhere (internal API).  If you let me know I'll make it so

2. Unit Tests pass -- But they don't actually hit BQ.  Do we know if this gets hit in integration tests?

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by mmorgantaylor** [comment]
re: .conf setting, my humble opinion is that it's a good idea to rename the billing config variable to be more precise, and to reflect the substantive change to the BQ table.

**Comment by davidangb** [comment]
@kcibul if you need something to sanitize SQL outside of bind params: https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala#L54

**Review by mmorgantaylor — APPROVED** [review]

**Comment by kcibul** [comment]
jenkins retest

**Comment by kcibul** [comment]
jenkins retest

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/SubmissionCostService.scala [hunk 1]
```diff
@@ -80,14 +80,7 @@ class SubmissionCostService(tableName: String, serviceProject: String, billingSe
...
 
-    val namespaceParam =
-      new QueryParameter()
-        .setParameterType(stringParamType)
-        .setParameterValue(new QueryParameterValue().setValue(workspaceNamespace))
-
-    val queryParameters: List[QueryParameter] = List(namespaceParam)
-
-    executeBigQuery(querySql, queryParameters) map { result =>
+    executeBigQuery(querySql, List.empty) map { result =>
```
Comment (kshakir) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/SubmissionCostService.scala:83]: Out of curiosity, could you leave a comment on what this is doing? I see that it used to be something and now is `Nil`.

**Review by kshakir — APPROVED** [review]
Not an expert on what was there before (we're new owners...) but this LGTM if it works for you.

**Comment by kshakir** [comment]
As many comments as you have time to leave would be greatly appreciated!

**Pull request closed without merging**